backend/src/routes/auth.ts

```diff
@@ -16,6 +16,14 @@ export const registerHandler = (req: any, res: any) => {
         email: email,
         password: password
       };
+      if (password.length < 9){
+        res.send("password minimum 9 characters")
+        return;
+      }
+      if (!/\d/.test(password) || /^\d+$/.test(password)){
+        res.send("require alphanumeric password")
+        return;
```

Comment:
so there are 2 minor issues with this if statement:
1. if you want both the things being checked to be true, you need to use && not ||, the reason is if you use ||, if either the first or the second check being true will pass the validation 
2.  `/\d/` and `/^\d+$/` might not cover all cases.
 - the first regex will test if there is at least 1 number being matched. i.e. even this password satisfies the check `()!*@)#(*999` according to the first regex
   - the second regex only checks if **all** the values from start to end are only numbers. i.e. `12139081230` as a password
 
if alpha numeric values is what you want i.e. you don't want symbols then a potential regex could be
`/^[a-zA-Z0-9]+$/`
or
`/^(\w|\d)+$/`

Reply:
I think maybe in comments above write down the exact parameters on what you think the password check should include.

should the password be >9 but <n characters?
do we want there to be symbols? if so are we forcing them to include at least 1 symbol?